banyand/metadata/schema/index.go

```diff
@@ -93,6 +96,9 @@ func (e *etcdSchemaRegistry) GetIndexRule(ctx context.Context, metadata *commonv
 	if err := e.get(ctx, formatIndexRuleKey(metadata), &entity); err != nil {
 		return nil, err
 	}
+	if entity.Metadata.Id == 0 {
+		return nil, errors.WithMessagef(ErrEmptyIndexRuleID, "index rule: %v", metadata)
```

Comment:
Shall we return grpc status code since it it API related

Reply:
Make sense. I will introduce `DataLoss` to indicate this kind of data corruption.